### PR TITLE
Add support for soak jobs to restart cluster in kubetest

### PR DESCRIPTION
**Pull request by krzyzacy**
ref https://github.com/kubernetes/test-infra/issues/3779

/hold
probably a good time to add some more tests to kubetest

/assign @BenTheElder @fejta

### CODE REVIEW

**Comment by krzyzacy** [comment]
gcloud command works, I'll flip some existing soak jobs to prow to verify this tomorrow.

**Comment by krzyzacy** [comment]
http://prow.k8s.io/log?job=ci-kubernetes-soak-gci-gce-stable1&id=16

seems working now, we can start review the code
/lint

File: kubetest/main.go [hunk 1]
```diff
@@ -140,6 +142,8 @@ func defineFlags() *options {
 	flag.StringVar(&o.stage.dockerRegistry, "registry", "", "Push images to the specified docker registry (e.g. gcr.io/a-test-project)")
 	flag.StringVar(&o.save, "save", "", "Save credentials to gs:// path on --up if set (or load from there if not --up)")
 	flag.BoolVar(&o.skew, "skew", false, "If true, run tests in another version at ../kubernetes/hack/e2e.go")
+	flag.BoolVar(&o.soak, "soak", false, "If jobs runs as soak mode")
```
Comment (BenTheElder) [inline kubetest/main.go]: Nit: "If true, job runs in soak mode" is probably more consistent with the other usage strings.

**Comment by krzyzacy** [comment]
My English is obviously not the best
/hold cancel
gonna pop some more Jenkins jobs

**Pull request merged**